### PR TITLE
Reorganization and Addition: Move TLS types and config out of the outputs and support server options.

### PR DESCRIPTION
**Pull request by ph**
When working on the TLS TCP it was a bit strange to import a
package coming from the outputs; this commit addresses a few things:

- Move the `outputs/tls.go` and `transport/tls.go` into the common
under the transport folder.
- Add shims to make sure we keep backward compatibility on anything that
could be using theses classes.
- Extract common logic code to be reusable.
- Add inverse mapper for TLSVersion and tlsCiphersuite, to give a uint
and get the human string.
- Add a new `ServerConfig` config struct.

*This is a light refactoring, mostly moving code and adding a few
tests.

### CODE REVIEW

**Comment by ph** [comment]
I've rekick some Travis job which hit an http timeout when installing apt package.

**Comment by ph** [comment]
@kvch I've update the original PR with the new options: _client_authentication_.`, lets review this one first.

**Comment by ph** [comment]
jenkins test this please

File: libbeat/common/transport/tlscommon/types.go [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+
+func (m *tlsClientAuth) Unpack(in interface{}) error {
+	if in == nil {
+		*m = tlsClientAuthRequired
+		return nil
+	}
+
+	s, ok := in.(string)
+	if !ok {
+		return fmt.Errorf("client authentification must be an identifier")
```
Comment (kvch) [inline libbeat/common/transport/tlscommon/types.go]: authentication

File: libbeat/common/transport/tlscommon/types.go [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+	}
+
+	mode, found := tlsClientAuthTypes[s]
+	if !found {
+		return fmt.Errorf("unknown client authentification mode'%v'", s)
```
Comment (kvch) [inline libbeat/common/transport/tlscommon/types.go]: authentication

**Comment by ph** [comment]
@kvch Updated with the requested changes, I also added `authenfication` to my abolish.vim exception.

**Pull request merged**